functions-1/add-prefix-suffix-ngram-udfs.md

```diff
@@ -0,0 +1,141 @@
+---
+description: This section contains reference documentation for the add prefix, suffix and ngram UDFs.
+---
+
+# Add Prefix, Suffix & Ngram UDFs
+
+Provides prefix, suffix and ngram functionality for advanced string manipulation and text matching. These UDFs enable efficient text search and pattern matching capabilities, especially useful for wildcard searches and text prefiltering to avoid full table scans.
+
+## What is N-gram
+
+N-gram breaks down text into overlapping sequences of N characters. For example:
+
+**"Apache Pinot"** with 2-grams produces:
+- `Ap`, `pa`, `ac`, `ch`, `he`, `e `, ` P`, `Pi`, `in`, `no`, `ot`
+
+When matching `ngram(col, "ap")`, it searches for the substring "ap" within the generated n-grams.
+When matching `ngram(col, 'apache')`, it generates n-grams from "apache" (ap, pa, ac, ch, he) and matches against the column's n-grams.
+
+## When to Use N-gram
+
+N-grams are particularly useful for:
+
+- **Text matching** - Fast substring search
+- **Wildcard queries** - Alternative to `REGEXP_LIKE` (which requires full scan)
+- **Prefix matching** - More efficient than O(N * length of string) operations
+- **Prefiltering** - Avoid full table scans by using n-gram indexing
+
+### Performance Comparison
+- `REGEXP_LIKE`: Full scan O(N)
+- `Prefix matching`: O(N * length of string)
+- `Wildcard matching`: O(N * length)
+- **N-gram approach**: Enables prefiltering to avoid scanning all rows
+
+## Signature
+
+> NGRAM(col, n)
+>
+> PREFIX(col, prefixString)
+>
+> SUFFIX(col, suffixString)
+
+### `col`
+The input string column or expression to process.
+
+### `n`
+For n-gram operations, the length of each gram (typically 2 for bigrams, 3 for trigrams).
+
+### `prefixString` / `suffixString`
+The string to add as prefix or suffix to the input.
+
+## Usage Examples
+
+### N-gram Generation
+
+```sql
+SELECT NGRAM('Apache Pinot', 2) AS bigrams
+FROM myTable
+```
+
+| bigrams |
+| ------- |
+| Ap,pa,ac,ch,he,e ,  P,Pi,in,no,ot |
+
+```sql
+SELECT NGRAM('fast', 3) AS trigrams
+FROM myTable
+```
+
+| trigrams |
+| -------- |
+| fas,ast |
+
+### Prefix Operations
```

Comment:
remove this - is this unrelated.

Reply:
not sure I follow, why is this unrelated? we also want to doc for prefix?